rholang/src/main/scala/coop/rchain/rholang/interpreter/Reduce.scala

```diff
@@ -148,56 +156,77 @@ object Reduce {
       * @param par
       * @return
       */
-    override def eval(par: Par)(implicit env: Env[Par]): M[Unit] = {
-      def handle[A](eval: (A => M[Unit]))(a: A): M[Unit] =
-        eval(a).handleError((e: InterpreterError) => {
+    override def eval(par: Par)(implicit env: Env[Par], rand: Blake2b512Random): M[Unit] = {
+      val filteredExprs = par.exprs.filter { expr =>
+        expr.exprInstance match {
+          case _: EVarBody    => true
+          case _: EEvalBody   => true
+          case _: EMethodBody => true
+          case _              => false
+        }
+      }
+      val starts = Vector(par.sends.size,
+                          par.receives.size,
+                          par.news.size,
+                          par.matches.size,
+                          par.bundles.size,
+                          filteredExprs.size)
+        .scanLeft(0)(_ + _)
+      def handle[A](eval: (A => (Env[Par], Blake2b512Random) => M[Unit]), start: Int)(
+          ta: (A, Int)): M[Unit] = {
+        val newRand =
+          if (starts(6) == 1)
```

Comment:
If I understand correctly `starts(6)` yields the number of rholang exprssions right? Why do you do that in the `handle` method which will never be invoked with `Expr` (I understand that [this part](https://github.com/rchain/rchain/pull/1065/commits/331badc4a6cea1df6b80330cb2235e0959b84f42#diff-2e087015257005bc83676b5d76c6a366R194) is called on expressions).

Reply:
No, `starts(6)` is the total number of things in the par that need random state.